transcom/mymove
# B 21061 display sit authorized end dates int

## [B-21061](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1009581)

## Summary

Fixes payment request preview on past SITs never being possible. Introduces proper authorized end date tracking for SIT, making the `CalculateSITAuthorizedEndDate` function private and nested only under "SIT Summaries". Fixed an issue with unauthorized SIT still calculating a proper authorized end date, it now feeds the last known authorized SIT date if unauthorized.

Previously, the SIT authorized end date would only ever be functional for "Current/Future SIT". Calculating the authorized end date on past SITs would always be incorrect as it would not properly track all days used and if it is going past the allowance or not. This allowed "negative" authorized days

## How to test

- Open up a move under the prime sim
- Create a new HHG shipment
- Approve the shipment as a TOO
- Go back to Prime and create a PAST Origin SIT on that new shipment (Entry date May 1, departure May 7)
- Create a PAST destination SIT (First available delivery date May 1, Entry date June 1, departure September 1)
- Login as TOO and approve both
- As TOO, preview the “Previously used SIT section”
- You should see the following snippet calculations
 ![Pasted Graphic](https://github.com/user-attachments/assets/67765c24-bc43-4562-8368-eccc5aeaa89c)

- As the tester, please now verify it via https://www.timeanddate.com/date/durationresult.html

### Edge cases
Edge cases resolved by pull request are as follows:
- SIT created when there are no more authorized days giving an authorized end date in the future, when it should be in the past
- TIO reviewing payment requests for past SITs never functioning (Backend would panic due to nil ref for a hard coded "CurrentSIT" check)

## 508 compliance check
![Pasted Graphic 1](https://github.com/user-attachments/assets/8d82c25a-cbf6-4516-b8d1-431a54c36030)
